src/librustc_typeck/collect.rs

```diff
@@ -1632,7 +1632,7 @@ fn predicates_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredicates<'_> {
         // prove that the trait applies to the types that were
         // used, and adding the predicate into this list ensures
         // that this is done.
-        let span = tcx.def_span(def_id);
+        let span = tcx.sess.source_map().def_span(tcx.def_span(def_id));
```

Comment:
Ugh I think I forgot to open an issue about this after commenting... on one of @llogiq's PRs, maybe? `SourceMap::def_span` should be renamed ASAP to something like `guess_head_span`, the confusion factor is getting out of hand.

Reply:
Also I don't want to add more uses of the hacky approach without FIXMEs pointing to an issue about introducing something like `def_head_span` which would point to e.g. `trait Foo` or `fn bar` even cross-crate, and without guessing from source code after the fact.

Reply:
https://github.com/rust-lang/rust/pull/69866